openedx/adg/lms/applications/admin.py

```diff
@@ -119,14 +124,54 @@ class WorkExperienceAdmin(admin.ModelAdmin):
     search_fields = ('name_of_organization', 'job_position_title',)
 
 
-class PrerequisiteCourseAdmin(admin.TabularInline):
-    model = PrerequisiteCourse
+class MultilingualCourseInlineFormset(BaseInlineFormSet):
+    """
+    Inline formset for multilingual courses
+    """
+
+    def clean(self):
+        super(MultilingualCourseInlineFormset, self).clean()
+
+        course_group_lang_count = defaultdict(lambda: 0)
+        store = modulestore()
 
+        for form in self.forms:
+            if not form.cleaned_data:
+                continue
 
-@admin.register(PrerequisiteCourseGroup)
-class PrerequisiteCourseGroupAdmin(admin.ModelAdmin):
-    inlines = [PrerequisiteCourseAdmin, ]
-    list_display = ('name', 'prereq_course_count', 'open_prereq_courses_count',)
+            course_info = store.get_course(form.instance.course.id)
+            if not course_info.language:
+                raise ValidationError(_('Please add language of the course from studio.'))
+
+            course_group_lang_count[course_info.language] += 1
+            if course_group_lang_count[course_info.language] > 1:
+                raise ValidationError(
+                    _(
+                        'You cannot add 2 or more courses with same language.'
+                        ' Add courses of different language or change language of a course from studio!'
+                    )
+                )
+
+
+class MultilingualCourseAdmin(admin.TabularInline):
+    """
+    Inline admin for Multilingual Courses
+    """
+
+    model = MultilingualCourse
+    formset = MultilingualCourseInlineFormset
+
+    def get_extra(self, request, obj=None, **kwargs):
+        """
+        Customized number of inline forms
+        """
+        return 0 if obj and obj.multilingual_courses.exists() else 2
```

Comment:
Actually the method name suggests what it is and the method is also a method of super class, So I thought it would be better to avoid a constant for this, But sure your idea is what I thought initially as well. Will update this one.